#!/usr/bin/env python
# Path smoother configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t
PACKAGE = "nav2_smac_planner"

gen = ParameterGenerator()

gen.add("tolerance", double_t, 0, "Smoothing tolerance", 1e-10, 0.0, 10.0)
gen.add("max_iterations", int_t, 0, "Smoothing maximum number of iterations", 1000, 1, 1000000)
gen.add("w_data", double_t, 0, "Weight to apply data ???", 0.2, 0.0, 1000.0)
gen.add("w_smooth", double_t, 0, "Weight to apply smooth ???", 0.3, 0.0, 1000.0)
gen.add("do_refinement", bool_t, 0, "Whether to recursively run the smoother refinement_num times on the results from prior runs to refine the results further", True)
gen.add("refinement_num", int_t, 0, "Number of times to refine the result if do_refinement is true", 2, 1, 1000)

exit(gen.generate(PACKAGE, "smac_planner_hybrid", "Smoother"))
